python/lsst/obs/base/cameraMapper.py

```diff
@@ -336,56 +337,67 @@ def stdClosure(item, dataId, mapper=weakref.proxy(self), mapping=mapping):
-
-                    # Set up metadata versions
-                    if name == "exposures" or name == "images":
-                        expFunc = "map_" + datasetType  # Function name to map exposure
-                        mdFunc = expFunc + "_md"       # Function name to map metadata
-                        bypassFunc = "bypass_" + datasetType + "_md"  # Func name to bypass daf_persistence
-                        if not hasattr(self, mdFunc):
-                            setattr(self, mdFunc, getattr(self, expFunc))
-                        if not hasattr(self, bypassFunc):
-                            setattr(self, bypassFunc,
-                                    lambda datasetType, pythonType, location, dataId:
-                                    afwImage.readMetadata(location.getLocations()[0]))
-                        if not hasattr(self, "query_" + datasetType + "_md"):
-                            setattr(self, "query_" + datasetType + "_md",
-                                    getattr(self, "query_" + datasetType))
-
-                        subFunc = expFunc + "_sub"  # Function name to map subimage
-                        if not hasattr(self, subFunc):
-                            def mapSubClosure(dataId, write=False, mapper=weakref.proxy(self),
-                                              mapping=mapping):
-                                subId = dataId.copy()
-                                del subId['bbox']
-                                loc = mapping.map(mapper, subId, write)
-                                bbox = dataId['bbox']
-                                llcX = bbox.getMinX()
-                                llcY = bbox.getMinY()
-                                width = bbox.getWidth()
-                                height = bbox.getHeight()
-                                loc.additionalData.set('llcX', llcX)
-                                loc.additionalData.set('llcY', llcY)
-                                loc.additionalData.set('width', width)
-                                loc.additionalData.set('height', height)
-                                if 'imageOrigin' in dataId:
-                                    loc.additionalData.set('imageOrigin',
-                                                           dataId['imageOrigin'])
-                                return loc
-                            setattr(self, subFunc, mapSubClosure)
-                        if not hasattr(self, "query_" + datasetType + "_sub"):
-                            def querySubClosure(key, format, dataId, mapping=mapping):
-                                subId = dataId.copy()
-                                del subId['bbox']
-                                return mapping.lookup(format, subId)
-                            setattr(self, "query_" + datasetType + "_sub", querySubClosure)
+                    def setMethods(suffix, mapImpl=None, bypassImpl=None, queryImpl=None):
```

Comment:
I feel like this whole subPolicy block that you've cleanedup/added (basically most of the contents of this inner for loop) should be pulled into a separate `_addPolicies` method. `__init__` is really long as it stands, and this could help break it up (I didn't even see that this was all in a double-for loop!).